lib/assets/javascripts/twine.coffee

```diff
@@ -69,13 +70,29 @@
       element = findOrCreateElementForNode(node)
       element.indexes = indexes
 
-    for type, binding of Twine.bindingTypes when definition = Twine.getAttribute(node, type)
-      element = findOrCreateElementForNode(node)
+    bindingConstructors = null
+    for attribute in node.attributes
+      type = attribute.name
+      type = type.slice(5) if type.slice(0, 5) == 'data-'
+      definition = attribute.value
+
+      constructor = Twine.bindingTypes[type]
+      continue unless constructor
+
+      bindingConstructors ?= []
+      if type == 'bind'
```

Comment:
Sorry im in mobile and can't comment on the right line somehow. Regarding line 76: slice makes an entire new string object. I recommend you test if the 5th character of he attribute is '-' before you do the slice. That's a much faster way of testing for high probability of it being a data- attribute.